### DISCUSSION TITLE
Expression

### DISCUSSION THREAD

**Discussion by kjkj223** [exanauts/ExaModels.jl#84] [Q&A] [answered]
Hi. I recently switched from Python to Julia because I heard it can solve optimization problems quickly on GPUs. 
Does ExaModels have a 'expression' function like Jump or Pyomo?

**Comment by sshin23** [comment, marked as answer]
No, it doesn't have that feature. Currently, a possible workaround is introducing additional variables and enforcing them as constraints

**Reply by kjkj223** [reply]
Thank you!

**Comment by jbcaillau** [comment]
hi @sshin23 @frapac *bravo* for the work on `ExaModels.jl`. Is anything like `JuMP` [function tracing](https://jump.dev/JuMP.jl/stable/manual/nonlinear/#Function-tracing) in the pipeline? Of course, this kind of approach is expected to have limitations for large expressions, check [this post](https://jump.dev/JuMP.jl/stable/manual/nonlinear/#Function-tracing).
